File: StructuralPattern/Flyweight/Java/Concretes/Plant.java
package Concretes;

import Interfaces.GameObject;

public class Plant implements GameObject {
    private final String type; // Intrinsic state
    private final String image;

    public Plant(String type, String image) {
        this.type = type;
        this.image = image;
    }

    @Override
    public void display(int x, int y) {
        System.out.println("Plant: " + type + " with image: " + image + " is displayed at (" + x + ", " + y + ")");
    }
}

File: StructuralPattern/Flyweight/Golang/Concretes/Plant.go
package Concretes

import (
	"fmt"
	"v/Interfaces"
)

/**
 * Struct Plant implementation Game Object
 */
type Plant struct {
    Type string // Intrinsic state
    Image string
}

// Construct a plant
func NewPlant(t string, image string) (iface.GameObject) {
	return &Plant{Type: t, Image: image}
}

// Override display
func (p *Plant) Display(x int, y int) {
	// Display the plant
	fmt.Println("Displaying Plant: ", p.Type, " at (", x, ", ", y, ")")
}
